test/unit/component.js

```diff
@@ -82,142 +85,344 @@ describe('Component class', () => {
+            await component.init(data);
+          } catch (error) {
+            assert.equal(error, setupError);
+          }
+          assert.notCalled(logNotFoundStub);
+        });
       });
     });
 
-    describe('get selectors()', () => {
-      it('returns classes formatted as css selectors for each component in manifest', () => {
-        const component = new Component({id: 1234});
-        component.typeKey = 'product';
-        assert.equal(component.selectors.product.product, '.shopify-buy__product');
-        assert.equal(component.selectors.option.option, '.shopify-buy__option-select');
+    describe('setupModel()', () => {
+      it('returns passed data', async () => {
+        const model = await component.setupModel({test: 'test'});
+        assert.deepEqual(model, {test: 'test'});
+      });
+
+      it('fetches data if data was not passed', async () => {
+        component.fetchData = sinon.stub().resolves({test: 'fetchData'});
+        const model = await component.setupModel();
+        assert.calledOnce(component.fetchData);
+        assert.deepEqual(model, {test: 'fetchData'});
       });
     });
 
-    describe('get googleFonts()', () => {
-      it('returns google fonts for each component in manifest', () => {
-        const config = {
-          options: {
-            product: {
-              googleFonts: ['lol'],
-            },
-            option: {
-              googleFonts: ['bar'],
-            },
-          },
-        };
-        const component = new Component(config);
-        component.typeKey = 'product';
-        assert.deepEqual(component.googleFonts, ['lol', 'bar']);
+    describe('updateConfig()', () => {
+      it('delegates to updater', () => {
+        const updateConfigStub = sinon.stub(component.updater, 'updateConfig').returns('updated');
+        const returnVal = component.updateConfig('test');
+        assert.calledWith(updateConfigStub, 'test');
+        assert.equal(returnVal, 'updated');
+        updateConfigStub.restore();
       });
     });
 
-    describe('get viewData()', () => {
-      it('returns model and some other properties', () => {
-        const component = new Component({id: 1234});
-        component.typeKey = 'product';
-        component.model = {test: 'lol'};
-        assert.equal(component.viewData.test, 'lol');
-        assert.equal(component.viewData.classes.product.product, 'shopify-buy__product');
+    describe('destroy()', () => {
+      it('delegates to view', () => {
+        const destroyStub = sinon.stub(component.view, 'destroy');
+        component.destroy();
+        assert.calledOnce(destroyStub);
+        destroyStub.restore();
       });
     });
 
-    describe('init()', () => {
-      it('assigns model and initializes view', () => {
-        const component = new Component({id: 1234}, {
-          browserFeatures: {},
+    describe('getters', () => {
+      describe('name', () => {
+        it('returns name based on handle or id', () => {
```

Comment:
Since we're creating the initial component in the prototype's beforeEach, it is a bit hard to trace where the `1234` comes from. Could we put the id into it's own variable, and use that when we create the component in the beforeEach, and also reference it here for clarity?